assets/app/scripts/services/data.js

```diff
@@ -912,29 +916,29 @@ DataService.prototype.createStream = function(resource, name, context, opts, isR
       });
     }
 
-    this._resourceVersion(resource, context, data.resourceVersion || data.metadata.resourceVersion);
-    this._data(resource, context, data.items);
-    this._listCallbacks(resource, context).fire(this._data(resource, context));
-    this._listCallbacks(resource, context).empty();
-    this._watchCallbacks(resource, context).fire(this._data(resource, context));
+    this._resourceVersion(key, data.resourceVersion || data.metadata.resourceVersion);
+    this._data(key, data.items);
+    this._listCallbacks(key).fire(this._data(key));
+    this._listCallbacks(key).empty();
+    this._watchCallbacks(key).fire(this._data(key));
 
     // mark list op as complete
-    this._listInFlight(resource, context, false);
+    this._listInFlight(key, false);
 
-    if (this._watchCallbacks(resource, context).has()) {
-      var watchOpts = this._watchOptions(resource, context) || {};
+    if (this._watchCallbacks(key).has()) {
+      var watchOpts = this._watchOptions(key) || {};
       if (watchOpts.poll) {
-        this._watchInFlight(resource, context, true);
-        this._watchPollTimeouts(resource, context, setTimeout($.proxy(this, "_startListOp", resource, context), watchOpts.pollInterval || 5000));
+        this._watchInFlight(key, true);
+        this._watchPollTimeouts(key, setTimeout($.proxy(this, "_startListOp", key), watchOpts.pollInterval || 5000));
       }
-      else if (!this._watchInFlight(resource, context)) {
-        this._startWatchOp(resource, context, this._resourceVersion(resource, context));
+      else if (!this._watchInFlight(key)) {
+        this._startWatchOp(key, resource, context, this._resourceVersion(key));
       }
     }
   };
 
-  DataService.prototype._startWatchOp = function(resource, context, resourceVersion) {
-    this._watchInFlight(resource, context, true);
+  DataService.prototype._startWatchOp = function(key, resource, context, resourceVersion) {
```

Comment:
So Q. Since our `watch` will do a list first, do we need the keys to be unique for `https` vs `wss`?  I think we do not, it should be the same data either way if the rest of the url matches.